<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [ImageJ-bugs] [Bug 890] python broke after latest update
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:imagej-bugs%40imagej.net?Subject=Re%3A%20%5BImageJ-bugs%5D%20%5BBug%20890%5D%20python%20broke%20after%20latest%20update&In-Reply-To=%3Cbug-890-19-UeOc0bxmtp%40http.fiji.sc/bugzilla/%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="000227.html">
   <LINK REL="Next"  HREF="000245.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[ImageJ-bugs] [Bug 890] python broke after latest update</H1>
    <B>bugzilla at fiji.sc</B> 
    <A HREF="mailto:imagej-bugs%40imagej.net?Subject=Re%3A%20%5BImageJ-bugs%5D%20%5BBug%20890%5D%20python%20broke%20after%20latest%20update&In-Reply-To=%3Cbug-890-19-UeOc0bxmtp%40http.fiji.sc/bugzilla/%3E"
       TITLE="[ImageJ-bugs] [Bug 890] python broke after latest update">bugzilla at fiji.sc
       </A><BR>
    <I>Sat Aug 16 09:09:01 CDT 2014</I>
    <P><UL>
        <LI>Previous message: <A HREF="000227.html">[ImageJ-bugs] [Bug 890] python broke after latest update
</A></li>
        <LI>Next message: <A HREF="000245.html">[ImageJ-bugs] [Bug 890] python broke after latest update
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#241">[ date ]</a>
              <a href="thread.html#241">[ thread ]</a>
              <a href="subject.html#241">[ subject ]</a>
              <a href="author.html#241">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE><A HREF="http://fiji.sc/bugzilla/show_bug.cgi?id=890">http://fiji.sc/bugzilla/show_bug.cgi?id=890</A>

--- Comment #6 from Kaitlin Snider &lt;<A HREF="http://imagej.net/mailman/listinfo/imagej-bugs">kaitlin.snider87 at gmail.com</A>&gt; ---
I am still having this issue.  Tried the fix of downloading Jython 2.5.2 and
saving the lib in Fiji.app/jars.  I'm getting this error:

Started Batch_Script.py at Sat Aug 16 10:04:35 EDT 2014
Traceback (most recent call last):
  File &quot;D:\Google Drive\Fiji\Fiji.app\plugins\Batch_Script.py&quot;, line 10, in
&lt;module&gt;
    flist = os.listdir(dir1);
    at com.kenai.jaffl.provider.jffi.AsmRuntime.marshal(AsmRuntime.java:167)

    at org.python.posix.WindowsLibC$jaffl$0.stat$raw(Unknown Source)

    at org.python.posix.WindowsLibC$jaffl$0.stat(Unknown Source)

    at org.python.posix.BaseNativePOSIX.stat(BaseNativePOSIX.java:200)

    at org.python.posix.LazyPOSIX.stat(LazyPOSIX.java:207)

    at org.python.modules.posix.PosixModule.listdir(PosixModule.java:480)

    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

    at
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)

    at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)

    at java.lang.reflect.Method.invoke(Method.java:597)


java.lang.IncompatibleClassChangeError: java.lang.IncompatibleClassChangeError:
Found class com.kenai.jffi.InvocationBuffer, but interface was expected

    at org.python.core.PyException.fillInStackTrace(PyException.java:70)
    at java.lang.Throwable.&lt;init&gt;(Throwable.java:181)
    at java.lang.Exception.&lt;init&gt;(Exception.java:29)
    at java.lang.RuntimeException.&lt;init&gt;(RuntimeException.java:32)
    at org.python.core.PyException.&lt;init&gt;(PyException.java:46)
    at org.python.core.PyException.&lt;init&gt;(PyException.java:43)
    at org.python.core.Py.JavaError(Py.java:495)
    at org.python.core.Py.JavaError(Py.java:488)
    at
org.python.core.PyReflectedFunction.__call__(PyReflectedFunction.java:188)
    at
org.python.core.PyReflectedFunction.__call__(PyReflectedFunction.java:204)
    at org.python.core.PyObject.__call__(PyObject.java:387)
    at org.python.core.PyObject.__call__(PyObject.java:391)
    at org.python.pycode._pyx5.f$0(D:\Google
Drive\Fiji\Fiji.app\plugins\Batch_Script.py:20)
    at org.python.pycode._pyx5.call_function(D:\Google
Drive\Fiji\Fiji.app\plugins\Batch_Script.py)
    at org.python.core.PyTableCode.call(PyTableCode.java:165)
    at org.python.core.PyCode.call(PyCode.java:18)
    at org.python.core.Py.runCode(Py.java:1275)
    at
org.scijava.plugins.scripting.jython.JythonScriptEngine.eval(JythonScriptEngine.java:76)
    at org.scijava.script.ScriptModule.run(ScriptModule.java:175)
    at org.scijava.module.ModuleRunner.run(ModuleRunner.java:167)
    at org.scijava.module.ModuleRunner.call(ModuleRunner.java:126)
    at org.scijava.module.ModuleRunner.call(ModuleRunner.java:65)
    at
org.scijava.thread.DefaultThreadService$2.call(DefaultThreadService.java:164)
    at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
    at java.util.concurrent.FutureTask.run(FutureTask.java:138)
    at
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
    at
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
    at java.lang.Thread.run(Thread.java:662)
Caused by: java.lang.IncompatibleClassChangeError: Found class
com.kenai.jffi.InvocationBuffer, but interface was expected
    at com.kenai.jaffl.provider.jffi.AsmRuntime.marshal(AsmRuntime.java:167)
    at org.python.posix.WindowsLibC$jaffl$0.stat$raw(Unknown Source)
    at org.python.posix.WindowsLibC$jaffl$0.stat(Unknown Source)
    at org.python.posix.BaseNativePOSIX.stat(BaseNativePOSIX.java:200)
    at org.python.posix.LazyPOSIX.stat(LazyPOSIX.java:207)
    at org.python.modules.posix.PosixModule.listdir(PosixModule.java:480)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at
org.python.core.PyReflectedFunction.__call__(PyReflectedFunction.java:186)
    ... 19 more

This happens with a simple batch script:

#Name the directory MANUALLY write in the folder on the next line
dir1 = &quot;C:\\Users\\Stormcrow\\Google Drive\\Research\\Circ Rhythms\\Obrietan
Lab\\Techniques\\Analysis\\DRAQ ROI Analysis\\14-8-1 Optimized\\&quot;

#And check:
print(&quot;The folder is: &quot; + dir1);

#List all files in the directory
import os

flist = os.listdir(dir1);

#And check:
print(&quot;The files within this folder are: &quot;)
for fname in flist:
    print fname

#Run a plugin on each file
from ij import IJ

for fname in flist:
    imp = IJ.openImage(dir1+fname)
    imp.show()
#Insert Plugin below - use record macro if necessary to get &quot;Name&quot;
    i = IJ.getImage()
    IJ.run(&quot;DRAQ ROIs wo TWS&quot;)
    print(fname + &quot; completed processing&quot;, &quot;&quot;)

-- 
You are receiving this mail because:
You are the assignee for the bug.
-------------- next part --------------
An HTML attachment was scrubbed...
URL: &lt;<A HREF="http://imagej.net/pipermail/imagej-bugs/attachments/20140816/397d3a98/attachment.html">http://imagej.net/pipermail/imagej-bugs/attachments/20140816/397d3a98/attachment.html</A>&gt;
</PRE>


<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="000227.html">[ImageJ-bugs] [Bug 890] python broke after latest update
</A></li>
	<LI>Next message: <A HREF="000245.html">[ImageJ-bugs] [Bug 890] python broke after latest update
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#241">[ date ]</a>
              <a href="thread.html#241">[ thread ]</a>
              <a href="subject.html#241">[ subject ]</a>
              <a href="author.html#241">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://imagej.net/mailman/listinfo/imagej-bugs">More information about the Imagej-bugs
mailing list</a><br>
</body></html>
